db/Pos/DB/GState/Common.hs

```diff
@@ -69,29 +75,25 @@ writeBatchGState = dbWriteBatch' GStateDB
 -- Common getters
 ----------------------------------------------------------------------------
 
+type MonadBlockDBSafe m = (MonadDBRead m, BlockchainHelpers MainBlockchain)
+
 -- | Get current tip from GState DB.
 getTip :: MonadDBRead m => m HeaderHash
 getTip = maybeThrow (DBMalformed "no tip in GState DB") =<< getTipMaybe
 
--- | Get maximum seen chain difficulty (used to prevent improper rollbacks).
-getMaxSeenDifficulty :: MonadDBRead m => m ChainDifficulty
-getMaxSeenDifficulty =
-    maybeThrow (DBMalformed "no max chain difficulty in GState DB") =<<
-    getMaxSeenDifficultyMaybe
-
 -- | Get 'Block' corresponding to tip.
-getTipBlockGeneric
-    :: forall block header undo m.
-       MonadBlockDBGeneric header block undo m
-    => m block
-getTipBlockGeneric = getTipSomething "block" (dbGetBlock @_ @block)
+getTipBlock :: MonadDBRead m => m Block
+getTipBlock = getTipSomething "block" dbGetBlock
+
+-- | Returns header of block that was requested from Block DB.
+getHeader
```

Comment:
Why is it in `GState` module? It's not stored in GState DB, it's stored in Block DB.